quill-sql/src/main/scala/io/getquill/context/sql/Prepare.scala

```diff
@@ -32,21 +26,20 @@ object Prepare {
           other.show
       }
 
-    val (entity, insert) = ExtractEntityAndInsertAction(bindedAst)
-    val isInsert = insert.isDefined
-
-    val generated = if (isInsert) entity.flatMap(nameGeneratedColumn) else None
-
-    (sqlString, idents, generated)
+    (sqlString, idents, returning.map(prop => returningColumn(ast, prop)))
   }
 
-  private def nameGeneratedColumn(entity: Entity)(implicit n: NamingStrategy): Option[String] = {
+  private def returningColumn(ast: Ast, prop: String)(implicit n: NamingStrategy) = {
```

Comment:
Actually, since it touches the `NamingStrategy`, I think it is closer to `idiom` and `show` than to `RenameProperties` or `RenameAssigment`. Anyway, I will move the whole logic of renaming the returning property to its own class, I think it is cleaner this way.